src/dotnet-scaffolding/dotnet-scaffold-aspnet/ScaffolderBuilderExtensions.cs

```diff
@@ -0,0 +1,101 @@
+using Microsoft.DotNet.Scaffolding.TextTemplating;
+using Microsoft.DotNet.Tools.Scaffold.AspNet.Commands.Blazor.BlazorCrud;
+using Microsoft.DotNet.Tools.Scaffold.AspNet.Commands.MinimalApi;
+using Microsoft.DotNet.Tools.Scaffold.AspNet.Helpers;
+
+namespace Microsoft.DotNet.Scaffolding.Core.Builder;
+
+internal static class ScaffolderBuilderExtensions
+{
+    public static IScaffoldBuilder WithMinimalApiTextTemplatingStep(this IScaffoldBuilder builder)
+    {
+        var allT4Templates = new TemplateFoldersUtilities().GetAllT4Templates(["MinimalApi"]);
+        string? t4TemplatePath = null;
+        return builder.WithStep<TextTemplatingStep>(config =>
+        {
+            config.Step.SkipStep = false;
+            var step = config.Step;
+            var context = config.Context;
+            MinimalApiModel? minimalApiModel = null;
+            if (context.Properties.TryGetValue(nameof(MinimalApiModel), out var minimalApiModelObj) &&
+                minimalApiModelObj is MinimalApiModel)
+            {
+                minimalApiModel = minimalApiModelObj as MinimalApiModel;
+            }
+
+            ArgumentNullException.ThrowIfNull(minimalApiModel);
+            if (minimalApiModel.DbContextInfo.EfScenario)
+            {
+                t4TemplatePath = allT4Templates.FirstOrDefault(x => x.EndsWith("MinimalApiEf.tt", StringComparison.OrdinalIgnoreCase));
+            }
+            else
+            {
+                t4TemplatePath = allT4Templates.FirstOrDefault(x => x.EndsWith("MinimalApi.tt", StringComparison.OrdinalIgnoreCase));
+            }
+
+            var templateType = MinimalApiHelper.GetMinimalApiTemplateType(t4TemplatePath);
+
+            if (string.IsNullOrEmpty(t4TemplatePath) ||
+                string.IsNullOrEmpty(minimalApiModel.EndpointsPath) ||
+                templateType is null)
+            {
+                throw new InvalidOperationException("could not get minimal api template");
+            }
+
+            step.TemplatePath = t4TemplatePath;
+            step.TemplateType = templateType;
+            step.TemplateModel = minimalApiModel;
+            step.TemplateModelName = "Model";
+            step.OutputPath = minimalApiModel.EndpointsPath;
+        });
+        
+    }
+
+    public static IScaffoldBuilder WithBlazorCrudTextTemplatingStep(
+        this IScaffoldBuilder builder)
+    {
+        var allT4TemplatePaths = new TemplateFoldersUtilities().GetAllT4Templates(["BlazorCrud"]);
+        foreach (var templatePath in allT4TemplatePaths)
+        {
+            var templateName = Path.GetFileNameWithoutExtension(templatePath);
+            var templateType = BlazorCrudHelper.GetTemplateType(templatePath);
+            if (!string.IsNullOrEmpty(templatePath) && templateType is not null)
+            {
+                builder = builder.WithStep<TextTemplatingStep>(config =>
+                {
+                    config.Step.SkipStep = false;
```

Comment:
having to reset the `SkipStep` variable because if set to true, it remains true for cascading steps. So if I have 3 `TextTemplatingStep` steps chained together in Program.cs and set `TextTemplatingStep1.SkipStep = true`, the 2nd and 3rd steps default to true as well. Why is that? Assuming I'm setting it up wrong. Is it due to the singleton nature of initialization? @tlmii

Reply:
using `AddTransient` in the correct spots